### ISSUE TITLE
[Android] IsVisible property change won't work as expected via MVVM

### ISSUE THREAD

**Issue by tataelm** [dotnet/maui#8445]
### Description

When try to change the visibility of an UI Element in the same grid cell, change won't work on Android. It is working fine on Windows. Not tested with MacOs, iOS.

I have tested it on Android emulator and on my physical device. 

**Goal**
Create 2 labels inside same grid cell. Let's say Label1 and Label2. By default, Label1 is visible and Label2 is NOT visible. 
By clicking a button, Label1 should become invisible and Label2 should become visible and so on... 
**Bug**
On Android, visibility of Label2 doesn't change.

### Steps to Reproduce

1. Create a new MAUI project
2. Add NuGet Package CommunityToolkit.Mvvm (8.0.0-preview4)
3. Create ViewModel folder
4. Add BaseViewModel

```
public partial class BaseViewModel : ObservableObject
{

}
```

5. Add MainPageViewModel

```
public partial class MainPageViewModel : BaseViewModel
{
    [ObservableProperty]
    bool _isLabel1Showing;
    [ObservableProperty]
    bool _isLabel2Showing;


    public MainPageViewModel()
    {
        IsLabel1Showing = true;
        IsLabel2Showing = false;
    }


    [RelayCommand]
    public void ShowHideLabels()
    {
        IsLabel1Showing = !IsLabel1Showing;
        IsLabel2Showing = !IsLabel2Showing;
    }
}
```

6. Add Dependency Injection in MauiProgram.cs

```
public static MauiApp CreateMauiApp()
{
	var builder = MauiApp.CreateBuilder();
	builder
		.UseMauiApp<App>()
		.ConfigureFonts(fonts =>
		{
			fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
			fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
		});

       builder.Services.AddSingleton<MainPage>();
       builder.Services.AddSingleton<MainPageViewModel>();

       return builder.Build();
}
```

7. Inject MainPageViewModel in MainPage.xaml.cs
```
public MainPage(MainPageViewModel viewModel)
{
    InitializeComponent();
    BindingContext = viewModel;
}
```
8. Prepare XAML file
```
<ContentPage
    x:Class="MauiApp_VisibilityTest.MainPage"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:viewmodel="clr-namespace:MauiApp_VisibilityTest.ViewModel"
    x:DataType="viewmodel:MainPageViewModel">


    <Grid>
        <Grid.RowDefinitions>
            <RowDefinition />
            <RowDefinition Height="80" />
        </Grid.RowDefinitions>


        <Label
            Grid.Row="0"
            HeightRequest="80"
            IsVisible="{Binding IsLabel1Showing}"
            Text="Label 1"
            WidthRequest="100" />
        <Label
            Grid.Row="0"
            HeightRequest="80"
            IsVisible="{Binding IsLabel2Showing}"
            Text="Label 2"
            WidthRequest="100" />

        <Button
            Grid.Row="1"
            Command="{Binding ShowHideLabelsCommand}"
            HeightRequest="80"
            Text="Change Visibility"
            WidthRequest="200" />

    </Grid>

</ContentPage>
```
9. Run on Windows to see expected behaviour
10.  Run on Android to see the bug. Label1 acts as expected but Label2 never becomes visible.

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11 - API 30

**Comment by Symbai** [comment]
Dupe of #8044 wait for next update or use a nightly build.

**Comment by jsuarezruiz** [comment]
Should be fixed by #8136
